fvtest/porttest/si.cpp

```diff
@@ -982,7 +982,25 @@ TEST(PortSysinfoTest, sysinfo_test_sysinfo_set_limit_CORE_FILE)
 			}
 			finalSoftLimit = (originalSoftLimit > newHardLimit)? newHardLimit: originalSoftLimit;
 		} else { /* running as root */
-			const uint64_t newHardLimit = 257;
+			const uint64_t newHardLimit = descriptorLimit + 1;
+
+			/* Try setting hard limit below soft limit. This should fail even for root user. */
+			rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_HARD, newHardLimit);
+			if (0 == rc) {
+				outputErrorMessage(PORTTEST_ERROR_ARGS, "omrsysinfo_set_limit lowered hard limit below soft limit\n");
+				reportTestExit(OMRPORTLIB, testName);
+				return;
+			}
+
+			/* First lower soft limit. */
+			rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS, descriptorLimit);
+			if (0 != rc) {
+				outputErrorMessage(PORTTEST_ERROR_ARGS, "omrsysinfo_set_limit FAILED rc=%d\n", rc);
+				reportTestExit(OMRPORTLIB, testName);
+				return;
+			}
+
+			/* Lower hard limit. */
 			rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_HARD, newHardLimit);
```

Comment:
Line 988 sets the hard limit to `256 + 1 = 257`. As per the above comment, this line should lower the hard limit. But, it is setting the hard limit again to `256 + 1 = 257`, which won't lower it. Should `newHardLimit` be `descriptorLimit` here?
```suggestion
			rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_HARD, descriptorLimit);
```

Reply:
Line 988 is testing that we can't lower the hard limit below the soft limit. It is supposed to fail (if it doesn't, the test exits immediately). Although I should probably set the hard limit on line 1004 to 256 since it is valid to set the hard limit to >= soft limit